R/read_csv.R

```diff
@@ -52,253 +75,266 @@
         }
+      } else if (metadata$method == "optimize") {
+        point_estimates <- posterior::as_draws_matrix(draws[1,, drop=FALSE])[, variables]
       }
     }
   }
-  repaired_model_params <- repair_variable_names(variables)
-  if (!is.null(warmup_draws)) {
-    posterior::variables(warmup_draws) <- repaired_model_params
-  }
-  if (!is.null(post_warmup_draws)) {
-    posterior::variables(post_warmup_draws) <- repaired_model_params
-  }
+
   if (length(not_matching) > 0) {
     not_matching_list <- paste(unique(not_matching), collapse = ", ")
-    warning("The supplied csv files do not match in the following arguments: ", not_matching_list, "!")
+    warning("The supplied csv files do not match in the following arguments: ",
+            paste(not_matching_list, collapse = ", "), call. = FALSE)
   }
-  sampling_info$model_params <- repair_variable_names(sampling_info$model_params)
-  sampling_info$inv_metric <- NULL
-  list(
-    sampling_info = sampling_info,
-    inv_metric = inv_metric,
-    step_size = step_size,
-    warmup_draws = warmup_draws,
-    post_warmup_draws = post_warmup_draws,
-    warmup_sampler_diagnostics = warmup_sampler_diagnostics_draws,
-    post_warmup_sampler_diagnostics = post_warmup_sampler_diagnostics_draws
-  )
-}
 
-# FIXME: also parse the csv header
-read_optim_csv <- function(files) {
-  stopifnot(length(files) == 1)
-  csv_no_comments <- utils::read.csv(
-    files,
-    comment.char = "#",
-    colClasses = "numeric"
-  )
-  mat <- as.matrix(csv_no_comments)
-  colnames(mat) <- repair_variable_names(colnames(mat))
+  metadata$inv_metric <- NULL
+  metadata$lines_to_skip <- NULL
+  metadata$model_params <- repair_variable_names(metadata$model_params)
+  repaired_variables <- repair_variable_names(variables)
+  if (metadata$method == "sample") {
+    if (!is.null(warmup_draws)) {
+      posterior::variables(warmup_draws) <- repaired_variables
+    }
+    if (!is.null(post_warmup_draws)) {
+      posterior::variables(post_warmup_draws) <- repaired_variables
+    }
+    list(
+      metadata = metadata,
+      inv_metric = inv_metric,
+      step_size = step_size,
+      warmup_draws = warmup_draws,
+      post_warmup_draws = post_warmup_draws,
+      warmup_sampler_diagnostics = warmup_sampler_diagnostics_draws,
+      post_warmup_sampler_diagnostics = post_warmup_sampler_diagnostics_draws
+    )
+  } else if (metadata$method == "variational") {
+    metadata$model_params <- metadata$model_params[metadata$model_params != "lp__"]
+    metadata$model_params <- gsub("log_p__", "lp__", metadata$model_params)
+    metadata$model_params <- gsub("log_g__", "lp_approx__", metadata$model_params)
+    repaired_variables <- repaired_variables[repaired_variables != "lp__"]
+    repaired_variables <- gsub("log_p__", "lp__", repaired_variables)
+    repaired_variables <- gsub("log_g__", "lp_approx__", repaired_variables)
```

Comment:
I added these because we need to also do this renaming in `repaired_variables` (since these becomes the new names in the output) not just `metadata$model_params`.